openshift-kube-apiserver/authorization/minimumkubeletversion/minimum_kubelet_version.go

```diff
@@ -0,0 +1,86 @@
+package minimumkubeletversion
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/blang/semver/v4"
+	nodelib "github.com/openshift/library-go/pkg/config/node"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/auth/nodeidentifier"
+)
+
+var minimumKubeletVersion *semver.Version
+
+func SetMinimumKubeletVersion(version string) {
+	if version == "" {
+		return
+	}
+	v := semver.MustParse(version)
+	minimumKubeletVersion = &v
+}
+
+type minimumKubeletVersionAuth struct {
+	nodeIdentifier nodeidentifier.NodeIdentifier
+	nodeInformer   corev1informers.NodeInformer
+}
+
+func NewMinimumKubeletVersion(nodeIdentifier nodeidentifier.NodeIdentifier,
+	nodeInformer corev1informers.NodeInformer,
+) authorizer.Authorizer {
+
+	return &minimumKubeletVersionAuth{
+		nodeIdentifier: nodeIdentifier,
+		nodeInformer:   nodeInformer,
+	}
+}
+
+func (m *minimumKubeletVersionAuth) Authorize(ctx context.Context, attrs authorizer.Attributes) (authorizer.Decision, string, error) {
+	if minimumKubeletVersion == nil {
+		return authorizer.DecisionNoOpinion, "", nil
+	}
+
+	nodeName, isNode := m.nodeIdentifier.NodeIdentity(attrs.GetUser())
+	if !isNode {
+		// ignore requests from non-nodes
+		return authorizer.DecisionNoOpinion, "", nil
+	}
+
+	if len(nodeName) == 0 {
+		return authorizer.DecisionNoOpinion, fmt.Sprintf("unknown node for user %q", attrs.GetUser().GetName()), nil
+	}
+
+	// Short-circut if "subjectaccessreviews", or a "get" or "update" on the node object.
+	// Regardless of kubelet version, it should be allowed to do these things.
```

Comment:
What about node subresources? Are proxy updates meant to be short-circuited here too?